dbt/adapters/base/impl.py

```diff
@@ -274,7 +281,7 @@ def __init__(self, config, mp_context: SpawnContext) -> None:
         self._macro_resolver: Optional[MacroResolverProtocol] = None
         self._macro_context_generator: Optional[MacroContextGeneratorCallable] = None
         # this will be updated to include global behavior flags once they exist
-        self.behavior = []  # type: ignore
+        self.behavior = self.GLOBAL_BEHAVIOR_FLAGS  # type: ignore
```

Comment:
The flags should just be put directly here, or as a module level attribute and included here. We should avoid adding them  as a class variable; this would avoid confusion of `self.behavior` versus `self.GLOBAL_BEHAVIOR_FLAGS`.

Reply:
> or as a module level attribute and included here

Sorry I'm not following this suggestion, could you please elaborate?

Reply:
I would either move `GLOBAL_BEHAVIOR_FLAGS` up to line 101 or I would define the behavior flags directly:
```python
def __init__(self, config, mp_context: SpawnContext) -> None:
    ...
    self.behavior = [
        {
            "name": "require_batched_execution_for_custom_microbatch_strategy",
            "default": False,
            "docs_url": "https://docs.getdbt.com/docs/build/incremental-microbatch",
        }
    ]
```
This reduces the number of class variables on `BaseAdapter` and avoids folks accidentally using `self.GLOBAL_BEHAVIOR_FLAGS`, which would be incorrect (they don't account for user settings, they won't fire warnings, etc.).

Reply:
I'm picking up what you're putting down. I think we should keep it as a variable, but move it outside the class as you suggested. My worry with it being an inline specification, that is that if anyone is overriding the `__init__` in their implementation, they might need to pull in what the default builtins behavior flags are.

I also might rename it to something like `DEFAULT_BUILTIN_BEHAVIOR_FLAGS` as well to be more clear. Someone might misunderstand the `GLOBAL_BEHAVIOR_FLAGS` name and think it's a global accessor or something.

Reply:
Updated!